users/tests/test_unit_users_permissions.py

```diff
@@ -0,0 +1,117 @@
+            MagicMock(user=self.request.user)
+        )
+
+
+class TestAuthenticatedAdminPermissions(BaseHasPermissionsTestCase):
+    url = '/api/users/'
+
+    def test_users_viewset_returns_200_for_admin_type_user(self):
+        response = self.get_viewset_as(
+            self.url,
+            *self.create_user(CustomUser.UserType.ADMIN.name),
+        )
+        self.assertEqual(response.status_code, 200)
+
+    def test_users_viewset_returns_403_for_non_admin_type_user(self):
+        response = self.get_viewset_as(
+            self.url,
+            *self.create_user(CustomUser.UserType.EMPLOYEE.name),
+        )
+        self.assertEqual(response.status_code, 403)
+
+
+class TestAuthenticatedAdminOrOwnerUserPermissions(BaseHasPermissionsTestCase, BaseHasObjectPermissionsTestCase):
+    url = '/api/account/'
+    patch_object_owners = patch(
+        'users.models.CustomUser.objects',
+        BaseHasObjectPermissionsTestCase.object_owners,
+    )
+
+    def setUp(self):
+        self.permission = permissions.AuthenticatedAdminOrOwnerUser()
+        self.object_owners.clear()
+        self.request = MagicMock(user=MagicMock())
+        self.view = MagicMock()
+
+    def test_user_viewset_returns_200_for_object_owner_user(self):
+        user_and_password = self.create_user(CustomUser.UserType.EMPLOYEE.name)
+        url = self.url + str(user_and_password[0].pk) + '/'
+        response = self.get_viewset_as(
+            url,
+            *user_and_password,
+        )
+        self.assertEqual(response.status_code, 200)
+
+    def test_user_viewset_returns_403_for_not_object_owner_user(self):
+        user_and_password = self.create_user(CustomUser.UserType.EMPLOYEE.name)
+        url = self.url + str(user_and_password[0].pk) + '/'
+        response = self.get_viewset_as(
+            url,
+            *self.create_user(CustomUser.UserType.EMPLOYEE.name),
+        )
+        self.assertEqual(response.status_code, 403)
+
+    def test_user_viewset_returns_200_for_admin_type_user(self):
+        user_and_password = self.create_user(CustomUser.UserType.EMPLOYEE.name)
+        url = self.url + str(user_and_password[0].pk) + '/'
+        response = self.get_viewset_as(
+            url,
+            *self.create_user(CustomUser.UserType.ADMIN.name),
+        )
+        self.assertEqual(response.status_code, 200)
+
+    def test_user_viewset_returns_403_for_non_admin_type_user(self):
+        user_and_password = self.create_user(CustomUser.UserType.EMPLOYEE.name)
+        url = self.url + str(user_and_password[0].pk) + '/'
+        response = self.get_viewset_as(
+            url,
+            *self.create_user(CustomUser.UserType.EMPLOYEE.name),
+        )
+        self.assertEqual(response.status_code, 403)
+        response = self.get_viewset_as(
+            url,
+            *self.create_user(CustomUser.UserType.MANAGER.name),
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @patch_object_owners
```

Comment:
You can change it from decorator to context manager. It is used only once so it will be more readable in one place :)